### PR TITLE
fix(quantic): persisting case classification values

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4425](https://coveord.atlassian.net/browse/SFINT-4425)

### The Problem:
When QuanticCaseClassification doesn't return any suggestions and the user goes back & forth in the flow, the values  selected by the user are not persisted.

https://user-images.githubusercontent.com/86681870/157897575-8f601e6f-16ce-42b6-b4ee-661100c2b54f.mov

### The Fix:
The values of the user are now always persisted when going back and forth in the flow.

https://user-images.githubusercontent.com/86681870/157897746-308255ef-705b-4ed0-a394-513d8cb9bd57.mov

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js [hunk 1]
```diff
@@ -202,6 +202,9 @@ export default class QuanticCaseClassification extends LightningElement {
           this.showSelect();
         }
         this.setFieldValue(value, true);
+      } else {
+        const value = this.field.state.value;
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js:206]:
```suggestion
        this.setFieldValue(this.field.state.value);
```

**Review by nathanlb — APPROVED** [review]

File: packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js (hunk 1 above)
Comment (lbergeron) [inline packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js:206]: The fix itself seems correct, however I don't really understand why hitting the `back` button ends up executing this `else` block.

It seems many small fixes were made in `updateFieldState` and the function could now use some refactoring. My understanding is that, most of that logic depends on whether the field has already been selected. So maybe we should detect this condition first, and then do the rest of the logic.

File: packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js [hunk 2]
```diff
@@ -185,27 +185,17 @@ export default class QuanticCaseClassification extends LightningElement {
...
-      this.previousClassifications = this.classifications;
-      this.classifications =
-        this.field.state.suggestions.slice(
-          0,
-          Math.max(Number(this.maxSuggestions), 0)
-        ) ?? [];
+    if (!this.lockedState) {
       this.loading = this.field.state.loading;
-      if (this.newSuggestionsReceived) {
+      if (this.maxSuggestions > 0 && this.newSuggestionsReceived) {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js:190]: You could simplify a bit I think?

```suggestion
const hasNewSuggestions = this.maxSuggestions > 0 && this.newSuggestionsReceived;
const value = (hasNewSuggestions && this.isAutoSelectionNeeded) ? this.classifications[0].value : this.field.state.value;
this.setFieldValue(value, hasNewSuggestions);
this.handleSuggestionsDisplayBehaviour();
```

I am not sure what the 2nd parameter for the `setFieldValue` does, here I am assuming you want to pass it false if there was no new suggestion.

File: packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js [hunk 3]
```diff
@@ -432,4 +432,22 @@ export default class QuanticCaseClassification extends LightningElement {
       );
     }
   }
+
+  /**
+   * Handles the display behaviour of the suggestions.
+   * @returns {void}
+   */
+  handleSuggestionsDisplayBehaviour() {
```
Comment (erocheleau) [inline packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js:440]:
```suggestion
  handleDisplaySuggestions() {
```
Maybe?

**Review by erocheleau — APPROVED** [review]
Few comments, also is it possible to add a test for this?

**Review by lbergeron — APPROVED** [review]

**Pull request merged**